service/history/workflow/mutable_state_impl.go

```diff
@@ -2671,34 +2678,55 @@ func (ms *MutableStateImpl) loadBuildIds() ([]string, error) {
 	return searchAttributeValues, nil
 }
 
+// getReachabilityDeployment ignores DeploymentTransition.Deployment. If there is a pinned override,
+// it returns the override deployment. If there is an unpinned override or no override, it returns
+// execution_info.deployment, which is the last deployment that this workflow completed a task on.
+func (ms *MutableStateImpl) getReachabilityDeployment() *deploymentpb.Deployment {
+	versioningInfo := ms.GetExecutionInfo().GetVersioningInfo()
+	if override := versioningInfo.GetVersioningOverride(); override != nil {
+		if override.GetDeployment() != nil {
```

Comment:
I know this is basically the same but I kinda feel safer this way:
```suggestion
		if override.GetBehavior() == PINNED {
```

Reply:
totally, also more readable for people who don't know the VersioningOverride rule like we do.

Reply:
If we limit the tag to pinned only, then this func can become `getPinnedDeployment`